Automation/NSG/deploy-nsg.ps1

```diff
@@ -0,0 +1,185 @@
+
+    [Parameter(Mandatory=$False)]
+    [string]
+    $SourceAddressPrefixes,
+
+    [Parameter(Mandatory=$False)]
+    [string]
+    $DestinationAddressPrefies
+)
+
+# If template path provded, do not use default
+if ($TemplateFilePath) {
+    $DefaultTemplateFilePath = $TemplateFilePath
+}
+
+# Hash table for template deployment
+$templateParams = @{
+    tags = @{
+        "ownerEmail" = $OwnerEmail
+    }
+    networkSecurityGroups_name = @{
+        "securityGroupName" = $SecurityGroupName
+    }
+    securityRules = @{
+        "name" = $SecurityRuleName
+        "protocol" = $Protocol    
+        "sourcePortRange" = $SourcePortRange  
+        "destinationPortRange" = $DestinationPortRange
+        "sourceAddressPrefix" = $SourceAddressPrefix
+        "destinationAddressPrefix" = $DestinationAddressPrefix
+        "access" = $Access          
+        "priority" = $Priority
+        "direction" = $Direction
+        "sourcePortRanges" = $SourcePortRanges
+        "destinationPortRanges" = $DestinationPortRanges     
+        "sourceAddressPrefixes" = $SourceAddressPrefixes
+        "destinationAddressPrefixes" = $DestinationAddressPrefies
+    }
+
+# Example SSH security rule
+<#
+        "name" = SSH_Access;
+        "protocol" = "TCP";         
+        "sourcePortRange"= "*";   
+        "destinationPortRange"= "22";
+        "sourceAddressPrefix"= "97.113.101.80/32";
+        "destinationAddressPrefix"= "*"
+        "access"= "Allow";              
+        "priority"= 100;                 
+        "direction"= "Inbound";         
+        "sourcePortRanges"= "";         
+        "destinationPortRanges"= "";     
+        "sourceAddressPrefixes"= "";    
+        "destinationAddressPrefixes"= "";
+#>
+}
+
+# Login using Larissa's Login module & contains error handling)
+Import-Module ..\Login
+Login $TenantId $SPApplicationId $SPSecret $SubscriptionId
+
+# Error handling for general issues + resource group creation/update
+If (New-AzResourceGroup -Name $ResourceGroupName -Location $Location -Force) {
+    Write-Host -ForegroundColor Blue "`nResource Group created/updated successfully.`n"
+} Else {
+    Write-Host -ForegroundColor Red $_.Exception.Message
+    Write-Host -ForegroundColor Yellow "`nError creating/updating resource group. See above error message(s). Please double check all entered parameters and try again.`n"
+}
+
+# Error handling for general issues & nsg resource group deployment
+
+# Check if NSG already exists
+If (Get-AzNetworkSecurityGroup -Name $SecurityGroupName -ResourceGroupName $ResourceGroupName -ErrorAction Ignore) {
+    Write-Host -ForegroundColor Red "`nNetwork Security Group already exists, please try again with a different SecurityGroupName.`n"
+} Else {
+    # Deploy with template + parameters
+    If (New-AzResourceGroupDeployment -Name $DeploymentNames -ResourceGroupName $ResourceGroupName -TemplateFile $DefaultTemplateFilePath -TemplateParameterObject $templateParams -Verbose) {
```

Comment:
$DeploymentName, not $DeploymentNames - this caused a failure for me

Reply:
Fixed!